dvc/remote/local/__init__.py

```diff
@@ -569,3 +569,48 @@ def _get_unpacked_dir_names(self, checksums):
             if self.is_dir_checksum(c):
                 unpacked.add(c + self.UNPACKED_DIR_SUFFIX)
         return unpacked
+
+    def _needs_checkout(self, path_info, checksum, force):
+        if self._can_avoid_copy(path_info, checksum):
+            return False
+        return True
+
+    def _is_cache_type_copy(self):
+        if not self.path_info:
+            return False
+
+        try:
+            link_test_dir = self.path_info / "tmp_test"
+            self.makedirs(link_test_dir)
+
+            file1 = link_test_dir / uuid()
+            file2 = link_test_dir / uuid()
+
+            with open(str(file1), "wb") as fobj:
+                fobj.write(bytes(1))
+
+            self.link(file1, file2)
+
+            self.remove(link_test_dir)
```

Comment:
Also if something goes wrong, you shouldn't leave trash, so I would add this to except: block.

Reply:
done

Reply:
@pared Could we use `finally` instead of running remove twice?

Reply:
removed exception catching, to prevent cover-ups of bugs